enterprise/server/remote_execution/cgroup/cgroup_test.go

```diff
@@ -4,13 +4,83 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/block_io"
+	"github.com/buildbuddy-io/buildbuddy/server/util/proto"
 	"github.com/google/go-cmp/cmp"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/protobuf/testing/protocmp"
 
 	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	scpb "github.com/buildbuddy-io/buildbuddy/proto/scheduler"
 )
 
+func TestSettingsMap(t *testing.T) {
+	for _, test := range []struct {
+		name     string
+		settings *scpb.CgroupSettings
+		expected map[string]string
+	}{
+		{
+			name:     "nil",
+			settings: nil,
+			expected: map[string]string{},
+		},
+		{
+			name:     "all fields unset",
+			settings: nil,
+			expected: map[string]string{},
+		},
+		{
+			name: "all values set",
+			settings: &scpb.CgroupSettings{
+				CpuWeight:                proto.Int64(200),
+				CpuQuotaLimitUsec:        proto.Int64(400e3),
+				CpuQuotaPeriodUsec:       proto.Int64(100e3),
+				CpuMaxBurstUsec:          proto.Int64(50e3),
+				CpuUclampMin:             proto.Float32(12.34),
+				CpuUclampMax:             proto.Float32(98.76),
+				MemoryThrottleLimitBytes: proto.Int64(777e6),
+				MemoryLimitBytes:         proto.Int64(800e6),
+				MemorySoftGuaranteeBytes: proto.Int64(100e6),
+				MemoryMinimumBytes:       proto.Int64(50e6),
+				SwapThrottleLimitBytes:   proto.Int64(800e6),
+				SwapLimitBytes:           proto.Int64(1e9),
+				BlockIoLatencyTargetUsec: proto.Int64(100e3),
+				BlockIoWeight:            proto.Int64(300),
+				BlockIoLimit: &scpb.CgroupSettings_BlockIOLimits{
+					Riops: proto.Int64(1000),
+					Wiops: proto.Int64(500),
+					Rbps:  proto.Int64(4096e3),
+					Wbps:  proto.Int64(1024e3),
+				},
+			},
+			expected: map[string]string{
+				"cpu.weight":       "200",
+				"cpu.max":          "400000 100000",
+				"cpu.max.burst":    "50000",
+				"cpu.uclamp.min":   "12.34",
+				"cpu.uclamp.max":   "98.76",
+				"memory.high":      "777000000",
+				"memory.max":       "800000000",
+				"memory.low":       "100000000",
+				"memory.min":       "50000000",
+				"memory.swap.high": "800000000",
+				"memory.swap.max":  "1000000000",
+				"io.latency":       "279:8 target=100000",
+				"io.weight":        "279:8 300",
+				"io.max":           "279:8 riops=1000 wiops=500 rbps=4096000 wbps=1024000",
+			},
```

Comment:
can you add a test to verify that clamping works?

Reply:
added a test to make sure we return an error instead, since we're not clamping anymore